compiler/parser/RdrHsSyn.hs

```diff
@@ -2848,6 +2872,78 @@ mkArchiveChoice =
   where
     pureUnit = mkApp (mkUnqualVar $ mkVarOcc "pure") (noLoc $ ExplicitTuple noExt [] Boxed)
 
+validateException
+  :: Located RdrName
+  -> ExceptionBodyDecls
+  -> P ValidException
+validateException name ExceptionBodyDecls{..}
+  | length ebdMessage > 1 = report "Multiple 'message' declarations"
```

Comment:
maybe pattern match on the list directly instead of combining `length` asnd `listToMaybe`?